api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.time.LocalTime;
+
+/**
+ * A utility class that evaluates the concept ranges 
+ */
+public class ConceptReferenceRangeUtility {
+	
+	private static final Logger logger = LoggerFactory.getLogger(ConceptReferenceRangeUtility.class);
+	
+	public ConceptReferenceRangeUtility() {
+	}
+	
+	/**
+	 * This method evaluates the given criteria against the provided context.
+	 *
+	 * @param criteria the criteria string to evaluate
+	 * @param person person object containing variables to be used in the criteria
+	 * @return true if the criteria evaluates to true, false otherwise
+	 */
+	public static boolean evaluateCriteria(String criteria, Person person) {
+		if (criteria == null || criteria.isEmpty() || person == null) {
```

Comment:
It is misleading to evaluate to true simply because of null arguments. We should instead throw an exception.

Reply:
I don't believe so. Say for example a concept_ref_range is created without any criteria (assuming it doesn't need criteria), then saving this Obs won't be possible because the user will get an exception. The only bit I think we should throw an exception is when person is null. What do you think?

Reply:
We should not allow anyone to save a concept reference range without a criteria. Otherwise it would be a meaningless record. 😊

Reply:
Okay, got it.